src/components/dataset-query/dataset-query.js

```diff
@@ -1,31 +1,44 @@
-import { useState, useEffect } from 'react'
+import { useState } from 'react'
+import PropTypes from 'prop-types';
 import './dataset-query.scss'
+import LoadingElement from '../generic-elements/loading-element';
+
 
 const DatasetQuery = props => {
```

Comment:
Might also consider destructuring the props here.  I always think this is the clearest place to 'declare' what props a component is looking for.

`({ defaultQuery, onQueryDataset, ... }) => {`